test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java

```diff
@@ -323,6 +324,25 @@ public static Path getResourcesForClassesDirOrNull(Path classesDir, String name)
     }
 
     public static boolean isTestClass(String className, ClassLoader classLoader, Path testLocation) {
+        if (classLoader instanceof QuarkusClassLoader qcl) {
+            // this appears to be a more efficient and reliable way of performing this check
+            // the code after this IF block has an issue on Windows
+            final List<ClassPathElement> cpeList = qcl.getElementsWithResource(fromClassNameToResourceName(className),
+                    false);
+            if (!cpeList.isEmpty()) {
+                // if it's not empty, it's pretty much always a list with a single element
+                if (cpeList.size() == 1) {
+                    final ClassPathElement cpe = cpeList.get(0);
+                    return cpe.isRuntime() && cpe.getRoot().equals(testLocation);
```

Comment:
`cpe.getRoot()` theoretically could return `null`, so the `equals` call should be flipped. I can do that once the CI has finished.